src/InterventionBackend.php

```diff
@@ -294,27 +325,111 @@ public function getQuality()
     }
 
     /**
-     * @return int The width of the image
+     * Return dimensions as array with cache enabled
+     *
+     * @return array Two-length array with width and height
      */
-    public function getWidth()
+    protected function getDimensions()
     {
+        // Default result
+        $result = [0, 0];
+
+        // We can only cache dimensions from the source file (before any manipulations are performed),
+        // so ensure that the original image is un-loaded into memory.
+        // If a manipulation has been performed, we will return the dimensions from the already loaded
+        // image resource. In any case the 'variant' for these manipulations is unknown, so we couldn't
+        // cache those anyway.
+        $resource = $this->image;
+        if ($resource) {
+            return $this->getResourceDimensions($resource);
+        }
+
+        // Check if we have a container
+        $container = $this->getAssetContainer();
+        if (!$container) {
+            return $result;
+        }
+
+        // Check cache for unloded image
```

Comment:
Specifically, because the image can't change without the SHA also changing, the cases in which you'd want to discard the cache are rare enough that they can be handled with an explicit cache purge.

This means that the other logic isn't used unnecessarily: the normal flow will be that these fields are warmed on upload and then fetched from the cache for the rest of the object's lifetime.